### PR TITLE
modulesync 3.1.0

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 2.12.0

### CODE REVIEW

File: spec/acceptance/nodesets/archlinux-2-x64.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (ekohl) [inline spec/acceptance/nodesets/archlinux-2-x64.yml]: Should we keep this and the ec2 nodesets out? I've been wondering how useful they are.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,45 +1,18 @@
...
-    }
-  }.merge({})
-
-  c.before do
-    # avoid "Only root can execute commands as other users"
-    Puppet.features.stubs(root?: true)
-  end
-end
-
-shared_examples :compile, compile: true do
```
Comment (ekohl) [inline spec/spec_helper.rb]: How does this work? It looks like this always adds a test when a `describe` has `compile: true` and you're reducing coverage here.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,45 +1,18 @@
...
-    Puppet::Util::Log.newdestination(:console)
+if File.exist?(File.join(__dir__, 'default_module_facts.yml'))
+  facts = YAML.load(File.read(File.join(__dir__, 'default_module_facts.yml')))
+  if facts
+    facts.each do |name, value|
+      add_custom_fact name.to_sym, value
+    end
   end
-
-  c.default_facts = {
```
Comment (ekohl) [inline spec/spec_helper.rb]: I also see various tests that don't have an iteration of `on_supported_os` so I'd expect this to break something.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,45 +1,18 @@
...
-require 'puppetlabs_spec_helper/module_spec_helper'
-require 'rspec-puppet-facts'
-include RspecPuppetFacts
+# puppetlabs_spec_helper will set up coverage if the env variable is set.
+# We want to do this if lib exists and it hasn't been explicitly set.
+ENV['COVERAGE'] ||= 'yes' if Dir.exist?(File.expand_path('../../lib', __FILE__))
 
-RSpec.configure do |c|
-  c.mock_with :rspec
-  c.hiera_config = File.expand_path(File.join(__FILE__, '../fixtures/hiera.yaml'))
```
Comment (ekohl) [inline spec/spec_helper.rb]: Should also be removed.

**Comment by ekohl** [comment]
I've force pushed an update that's not complete yet, but it restores some functionality. That means it's not a pure modulesync. I like the shared example compile and I'm going to steal that. Also more classes need to be converted to rspec-puppet-facts.

**Comment by bastelfreak** [comment]
@ekohl any chance you can have a look again?

**Comment by bastelfreak** [comment]
this currently fails because the gnupg module is pretty broken  and we test a bunch of EOL operating systems :(

**Pull request merged**